magda-gateway/src/createBaseProxy.ts

```diff
@@ -75,5 +86,35 @@ export default function createBaseProxy(): httpProxy {
         });
     });
 
+    proxy.on("proxyReq", function(proxyReq, req, res) {
+        proxyReq.setHeader(MAGDA_TENANT_ID_HEADER, "undefined");
+
+        if (multiTenantsMode === true) {
+            const host = req.headers.host;
+            const domainName = new URI("http://" + host)
+                .hostname()
+                .toLowerCase();
+
+            if (domainName === magdaAdminPortalName) {
+                proxyReq.setHeader(
+                    MAGDA_TENANT_ID_HEADER,
+                    MAGDA_ADMIN_PORTAL_ID
+                );
+            } else {
+                const tenant = tenantsTable.get(domainName);
+                if (tenant !== undefined) {
+                    proxyReq.setHeader(MAGDA_TENANT_ID_HEADER, tenant.id);
+                } else {
+                    res.writeHead(500, { "Content-Type": "text/plain" });
```

Comment:
I think this should be a 4xx error. The client presumably did something wrong (used an invalid Host header), rather than the server broke.

Reply:
This is likely an admin error. For example, a website domain name resolves to the gateway ip address but the database does not have the entry of that tenant. Should it be status code 417 or a generic 400?
```
417 - Expectation Failed 
The web server responds with this status code when the expectation given in an Expect
request-header field could not be fulfilled by this server, or, if the server is a proxy and 
has clear evidence that the request could not be fulfilled by the next-hop server.
```

Reply:
fixed status code 400.